conu/backend/docker/container.py

```diff
@@ -475,3 +490,70 @@ def write_to_stdin(self, message):
             self.popen_instance.stdin.flush()
         except subprocess.CalledProcessError as e:
             raise ConuException(e)
+
+    def get_container_metadata(self):
+        """
+        Convert dictionary returned after docker inspect command into instance of ContainerMetadata class
+        :return: ContainerMetadata, container metadata instance
+        """
+
+        docker_metadata = self.get_metadata(refresh=True)
+
+        # format of Environment Variables from docker inspect:
+        # ['DISTTAG=f26container', 'FGC=f26']
+        env_variables = dict()
+        for env_variable in docker_metadata['Config']['Env']:
+            try:
+                env_variables.update({env_variable.split('=', 1)[0]: env_variable.split('=', 1)[1]})
+            except IndexError:
+                pass  # wrong format of environment variable
+
+        # format of image name from docker inspect:
+        # sha256:8f0e66c924c0c169352de487a3c2463d82da24e9442fc097dddaa5f800df7129
+        image = Image(docker_metadata['Image'].split(':')[1])
+
+        status = ContainerStatus.CREATED
+
+        if docker_metadata['State']['Status'] == 'created':
+            status = ContainerStatus.CREATED
+        elif docker_metadata['State']['Status'] == 'restarting':
+            status = ContainerStatus.RESTARTING
+        elif docker_metadata['State']['Status'] == 'running':
+            status = ContainerStatus.RUNNING
+        elif docker_metadata['State']['Status'] == 'removing':
+            status = ContainerStatus.REMOVING
+        elif docker_metadata['State']['Status'] == 'paused':
+            status = ContainerStatus.PAUSED
+        elif docker_metadata['State']['Status'] == 'exited':
+            status = ContainerStatus.EXITED
+        elif docker_metadata['State']['Status'] == 'dead':
+            status = ContainerStatus.DEAD
+
+        # format of Port mappings from docker inspect:
+        # {'12345/tcp': [{'HostIp': '0.0.0.0', 'HostPort': '123'}, {'HostIp': '0.0.0.0', 'HostPort': '1234'}]}
+        port_mappings = dict()
+        for key, value in docker_metadata['HostConfig']['PortBindings'].items():
+            for item in value:
+                if key in port_mappings.keys():
+                    port_mappings[key].append(int(item['HostPort']))
+                else:
+                    port_mappings.update({key: [int(item['HostPort'])]})
+
+        container_metadata = ContainerMetadata(
+            name=docker_metadata['Name'][1:],  # remove / at the beginning
+            identifier=docker_metadata['Id'],
+            labels=docker_metadata['Config']['Labels'],
+            command=docker_metadata['Config']['Cmd'],
+            creation_timestamp=docker_metadata['Created'],
+            env_variables=env_variables,
+            image=image,
+            exposed_ports=list(docker_metadata['Config']['ExposedPorts'].keys()),
```

Comment:
When there are no exposed ports on container, this field is missing in inspect and `get_container_metadata()` fails on KeyError